asserts/device_asserts_test.go

```diff
@@ -166,6 +166,34 @@ func (mods *modelSuite) TestDecodeRequiredSnapsAreOptional(c *C) {
 	c.Check(model.RequiredSnaps(), HasLen, 0)
 }
 
+func (mods *modelSuite) TestDecodeValidSnapNames(c *C) {
```

Comment:
seems we need a test that maybe focuses on just one of the fields but explores all the corner cases of validateSnapName